src/main/kotlin/com/lovelysystems/gradle/LSAws.kt

```diff
@@ -0,0 +1,105 @@
+package com.lovelysystems.gradle
+
+import org.gradle.api.Project
+import org.gradle.kotlin.dsl.getValue
+import org.gradle.kotlin.dsl.invoke
+import org.gradle.kotlin.dsl.provideDelegate
+import org.gradle.kotlin.dsl.registering
+import java.io.ByteArrayOutputStream
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.nio.file.StandardOpenOption
+
+private const val AWS_GROUP = "aws"
+private const val SSO_SESSION_NAME = "lovely-sso"
+private const val SSO_START_URL = "https://lovelysystems.awsapps.com/start"
+private const val SSO_REGION = "eu-central-1"
+
+fun Project.awsProject(profile: String) {
+
+    tasks {
+
+        /**
+         * Task to setup the AWS SSO session config if not already present. This is required to use AWS SSO and must be
+         * setup only once.
+         */
+        val ssoSessionSetup by tasks.registering {
+            group = AWS_GROUP
+            description = "Setup lovely-sso session config"
+
+            val userHome = System.getProperty("user.home")
```

Comment:
Probably move also into dolast{} since this part of code requires to run only in Gradles executition and not in project configuration phase. what do you think?